tests/kafkatest/tests/connect/templates/connect-distributed.properties

```diff
@@ -29,11 +29,6 @@ key.converter.schemas.enable={{ schemas|default(True)|string|lower }}
 value.converter.schemas.enable={{ schemas|default(True)|string|lower }}
 {% endif %}
 
-internal.key.converter=org.apache.kafka.connect.json.JsonConverter
```

Comment:
Not sure if we want to maintain any tests that validate overriding this continues to work. I think in practice it's probably never worked (and never been tested) with other values; but not sure if it is valuable to make sure including the settings still work.

I suspect it's not actually worth it, but for any other similar change, I'd normally push for compatibility validation. This case is a bit unusual since schema-d converters didn't ever fully work for internal stuff afaik, but we should think through and justify whether we think tests are necessary or not.

Reply:
We could beef up the `PluginsTest` unit test to ensure that overriding internal converters and/or their configs will still be picked up by the framework, but I think adding new integration/systems tests to make sure that things still work if/when that happens is unnecessary. If a different internal converter worked before these changes, and we can ensure that it will still be used in the same way by Connect after these changes, there's no reason it would stop working that would have anything to do with the changes made here.